sdks/go/pkg/beam/runners/universal/runnerlib/execute.go

```diff
@@ -94,11 +94,16 @@ func Execute(ctx context.Context, p *pipepb.Pipeline, endpoint string, opt *JobO
 	}
 	err = WaitForCompletion(ctx, client, jobID)
 
-	res, err := newUniversalPipelineResult(ctx, jobID, client)
-	if err != nil {
-		return presult, err
+	res, presultErr := newUniversalPipelineResult(ctx, jobID, client)
+	if presultErr != nil {
+		if err != nil {
+			return presult, errors.Wrap(err, presultErr.Error())
+		} else {
+			return presult, presultErr
+		}
+	} else {
+		presult = res
 	}
-	presult = res
 
 	return presult, err
 }
```

Comment:
```suggestion
}
```

cleanup of the old return that doesn't need to be preserved.